lib/internal/Magento/Framework/Escaper.php

```diff
@@ -318,6 +318,21 @@ public function escapeQuote($data, $addSlashes = false)
     }
 
     /**
+     * Escape "$" from text with the corresponding HTML code
+     *
+     * @param string $data
+     * 
+     * @return string mixed
+     */
+    public function escapeDollarSign($data)
```

Comment:
For now this is just some `Magento_Cms`-specific hack to overcome collision with JS templates syntax thus it should not be a part of `Magento_Framework`.

> to replace ${ with $&NoBreak;{

No need in HTML entity, just use unicode character. As we can simply do `str_replace('${', '$<proper code>{')` I don't think there is a real need to do it as separate method.

If we would need something more than one-liner, the class could be something like `\Magento\Cms\Model\PseudoEs6TemplateEscaper`.

Reply:
This collission happens with every field in the Magento backend, not only in `Magento_Cms` but with every text field in the backend https://github.com/magento/magento2/issues/10501#issuecomment-335301069.
According to me it doesn't make sense to put it in `Magento_Cms`

Reply:
`${` in product/category name would be pretty strange. Only those places needs to be fixed where we normally use some `{{ ... }}` now as a variable template.

My suggestion is: simply do `str_replace` in place, as soon as this logic this be identified frequently reusable (hope no) it can be refactored into separate class in whatever module or framework itself is suitable (now it should not go outside `Magento_Cms`, depending on use cases it may become a part of `Magento_Backend` or `Magento_Ui`).

Reply:
If someone sign up with a `${` inside the name, all the customer section in the backend is broken. Obviously it is not common, but we cannot trust user inputs. XSS by design!!!
`${` should NEVER be saved as is, so it makes sense to add the `Escaper' class to escape every user/admin inputs

Reply:
`Escaper` class  is not needed currently at all as I already said.

> If someone sign up with a `${` inside the name, all the customer section in the backend is broken.

This is a bit different story. In such case string replace does not make a lot of sense and also such situation never occurred before in reality. Please report it separately and we can discuss a fix for it then. I tend to forbidding such combination of symbols via validation.